### PR TITLE
(MODULES-4521) Use local copy of Puppet GPG keys

### PR DESCRIPTION
**Pull request by MikaelSmith**
Relying on the external GPG servers for getting GPG keys causes
intermittent test failures, and could also be an issue at sites that
limit network access. Mirroring the RedHat approach, include copies of
our GPG keys in the module and install them.

### CODE REVIEW

**Comment by highb** [comment]
@MikaelSmith Have you run this through a frankenbuild for PE?

**Comment by MikaelSmith** [comment]
No, only FOSS beaker tests. Can you run one?

**Comment by MikaelSmith** [comment]
Testing with `./frankenbuilder 2017.1 --install --vmpooler --upgrade-from 2016.4.2 --agent-upgrade --agent=debian-7-x86_64 --preserve_hosts=always --puppet_agent_pr 216`.

**Comment by MikaelSmith** [comment]
```
      Test Suite: pre_suite @ 2017-03-10 15:32:56 -0800

      - Host Configuration Summary -


              - Test Case Summary for suite 'pre_suite' -
       Total Suite Time: 1301.33 seconds
      Average Test Time: 433.78 seconds
              Attempted: 3
                 Passed: 3
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 3

      - Specific Test Case Status -

Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
Pending Tests Cases:
```

File: manifests/osfamily/debian.pp [hunk 1]
```diff
@@ -79,17 +79,42 @@
       }
     }
 
+    $legacy_keyname = 'DEB-GPG-KEY-puppetlabs'
```
Comment (MikaelSmith) [inline manifests/osfamily/debian.pp]: @underscorgan mind giving this a once-over to see if it's sane?

**Comment by MikaelSmith** [comment]
Yeah, I'll update that.

File: manifests/osfamily/redhat.pp [hunk 2]
```diff
@@ -48,10 +48,10 @@
     }
   }
 
-  $legacy_keyname = 'RPM-GPG-KEY-puppetlabs'
-  $legacy_gpg_path = "/etc/pki/rpm-gpg/${legacy_keyname}"
-  $keyname = 'RPM-GPG-KEY-puppet'
-  $gpg_path = "/etc/pki/rpm-gpg/${keyname}"
+  $legacy_keyname = 'GPG-KEY-puppetlabs'
+  $legacy_gpg_path = "/etc/pki/rpm-gpg/RPM-${legacy_keyname}"
```
Comment (MikaelSmith) [inline manifests/osfamily/redhat.pp]: I'm preserving the path here so that we don't write another copy for people who are already using this module.

**Comment by MikaelSmith** [comment]
Both FOSS beaker runs passed.
```
bundle exec beaker-hostgenerator redhat7-64mdcl-debian7-64default.a > hosts.yml
BEAKER_debug=true BEAKER_setfile=hosts.yml bundle exec rake beaker
bundle exec beaker-hostgenerator redhat7-64mdcl-redhat7-64default.a > hosts.yml
BEAKER_debug=true BEAKER_setfile=hosts.yml bundle exec rake beaker
```

**Comment by MikaelSmith** [comment]
Rerun on frankenbuilder also passed

```
      Test Suite: pre_suite @ 2017-03-10 16:49:55 -0800

      - Host Configuration Summary -


              - Test Case Summary for suite 'pre_suite' -
       Total Suite Time: 1370.77 seconds
      Average Test Time: 456.92 seconds
              Attempted: 3
                 Passed: 3
                 Failed: 0
                Errored: 0
                Skipped: 0
                Pending: 0
                  Total: 3

      - Specific Test Case Status -

Failed Tests Cases:
Errored Tests Cases:
Skipped Tests Cases:
Pending Tests Cases:
```

**Pull request merged**